### PR TITLE
Add small refactorings

### PR DESCRIPTION
**Pull request by tranzystorekk**
Hello, Stranger On The Internet ☺️

Here's a small set of changes that should simplify the code a fair bit:

- Simplify directory checking
- Destructure CLI args struct
- Simplify searching for the "-B" flag
- Remove redundant reference

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -44,21 +44,21 @@ fn run_cmake_command(build_dir: &str, args: &[String]) -> io::Error {
 }
 
 fn main() -> io::Result<()> {
-    let mut args = Args::parse();
+    let Args { mut build_directory, args } = Args::parse();
```
Comment (Qsionc) [inline src/main.rs:47]: This makes both build_directory and args variables mutable?

File: src/main.rs [hunk 2]
```diff
@@ -46,8 +46,7 @@ fn run_cmake_command(build_dir: &str, args: &[String]) -> io::Error {
 fn main() -> io::Result<()> {
     let Args { mut build_directory, args } = Args::parse();
 
-    if args.contains(&"-B".to_string()) {
-        let index = args.iter().position(|item| item == "-B").unwrap();
+    if let Some(index) = args.iter().position(|item| item == "-B") {
```
Comment (Qsionc) [inline src/main.rs:49]: AH i see, yes now i look at this its obvious.:D

File: src/main.rs (hunk 1 above)
Comment (tranzystorekk) [reply src/main.rs:47]: just `build_directory`

**Comment by Qsionc** [comment]
LGTM, +2

**Pull request merged**